.travis.yml

```diff
@@ -107,17 +108,55 @@ before_script:
     chmod +x ./cc-test-reporter
     ./cc-test-reporter before-build
   fi
+- export -f travis_fold
+- export -f travis_time_start
+- export -f travis_time_finish
 
 script:
-- if [[ "$PHPLINT" == "1" ]]; then find -L . -path ./vendor -prune -o -path ./node_modules -prune -o -name '*.php' -print0 | xargs -0 -n 1 -P 4 php -l; fi
-- if [[ "$PHPCS" == "1" ]]; then vendor/bin/phpcs -v --runtime-set ignore_warnings_on_exit 1; fi
-- if [[ "$CHECKJS" == "1" ]]; then grunt check:js; fi
-- if [[ "$COVERAGE" != "1" ]] && [[ ${TRAVIS_PHP_VERSION:0:1} == "5" || $TRAVIS_PHP_VERSION == hhv* ]]; then phpunit -c phpunit.xml; fi
-- if [[ "$COVERAGE" != "1" ]] && [[ ${TRAVIS_PHP_VERSION:0:1} == "7" || $TRAVIS_PHP_VERSION == "nightly" ]]; then phpunit -c phpunit.xml; fi
+# JavaScript checks
+- |
+  if [[ "$CHECKJS" == "1" ]]; then
+    travis_fold start "JavaScript.check" && travis_time_start
+    grunt check:js
+    travis_time_finish && travis_fold end "JavaScript.check"
+  fi
+# JavaScript tests
+- |
+  if [[ "$CHECKJS" == "1" ]]; then
+    travis_fold start "JavaScript.tests" && travis_time_start
+    yarn test
+    travis_time_finish && travis_fold end "JavaScript.tests"
+  fi
+# PHP Linting
+- |
+  if [[ "$PHPLINT" == "1" ]]; then
+    travis_fold start "PHP.check" && travis_time_start
+    find -L . -path ./vendor -prune -o -path ./node_modules -prune -o -name '*.php' -print0 | xargs -0 -n 1 -P 4 php -l
+    travis_time_finish && travis_fold end "PHP.check"
+  fi
+# PHP CS
+- |
```

Comment:
Folding is not needed here, again, you can change the command slightly to get a one-line output if all is ok (will show errors/warnings if those are encountered):
```bash
- if [[ "$PHPCS" == "1" ]]; then vendor/bin/phpcs -q --runtime-set ignore_warnings_on_exit 1; fi
```